### DISCUSSION TITLE
Rat brain registration HELP

### DISCUSSION THREAD

**Discussion by Byague** [ANTsX/ANTs#1639] [Q&A] [answered]
Dear community,

Im totally a newbie and unexpertise in this world of MRI preprocessing.

Im trying to register a rat brain template into my subject brain, to perform voxel analysis, but Im not able to perform the coregister propperly.

Maybe is a problem of the orientation, resolution, if I should applied the registration with a mask... or I dont know how to use the script properly.

Here is the input I use, and the rat brain atlas with its labels.

Im trying using directly antsRegistrationSyN.sh, without any previous step.

May someone could help me?

Balbino
[Rat1.zip](https://github.com/ANTsX/ANTs/files/13717841/Rat1.zip)

Link to template: https://www.nearlab.xyz/fischer344atlas

**Comment by ntustison** [comment]
First thing you would need to do is get a brain mask of your image.  Do you have that?

**Reply by Byague** [reply]
I upload here the mask!
[brainmask.zip](https://github.com/ANTsX/ANTs/files/13721062/brainmask.zip)

Thank you so much

**Reply by ntustison** [reply]
No problem.  But first, given that you're a newbie, it might be preferable to work with [ANTsPy](https://github.com/ANTsX/ANTsPy) or [ANTsR](https://github.com/ANTsX/ANTsR).  Do you have any experience with Python or R?

**Reply by Byague** [reply]
I have some experience with Python, but not too much. If it is possible I would like to work directly using ANTs scripts, but maybe is easy using ANTsPy

**Reply by ntustison** [reply]
Okay, that's fine.  We'll just go with the command line ANTs.  First issue is that your mask isn't as accurate as it should be.  Can you fix it?

<img width="1084" alt="Screenshot 2023-12-19 at 4 43 09 PM" src="https://github.com/ANTsX/ANTs/assets/324811/7f7d7951-a411-4745-b61e-47b3103837a5">

**Reply by Byague** [reply]
Here you have a better mask, output of a 3DUnet algorithm 
[mask2.zip](https://github.com/ANTsX/ANTs/files/13725574/mask2.zip)

**Reply by ntustison** [reply]
You can start with something like this:

```
template="Fischer344_nii_v4/Fischer344_template.nii"
template_mask="Fischer344_nii_v4/Fischer344_template_mask.nii"

specimen="AlzheimerTemplate.nii"
specimen_mask="brainmask.nii"

fixed_image="fixed.nii.gz"
moving_image="moving.nii.gz"

ImageMath 3 $fixed_image m $template $template_mask
ImageMath 3 $moving_image m $specimen $specimen_mask


antsRegistrationSyNQuick.sh -d 3 -f $fixed_image -m $moving_image -n 4 -o test
```

**Reply by Byague** [reply]
Here I share a link to the output... https://we.tl/t-2EJHperojl

**Reply by Byague** [reply]
I tried this time changing the fixed and moved in the command line to apply then the direct transform of the template labels to the moving image. First I use the labels of the template as downloaded, and achieved something horrible. I changed the orientation of the labels form IPL to LPI (in itksnap) and achieve something better (photo attached). May you know how to solve it? In the transformation I apply as interpolation methjod NearestNeighbor
![Screenshot from 2023-12-29 19-45-15](https://github.com/ANTsX/ANTs/assets/117201795/244da346-05f3-423b-88c9-5ad1c2425c33)

**Reply by ntustison** [reply]
A screenshot, generally, is not useful in diagnosing anything more than superficial issues.  You're going to need to upload the input data and the precise commands used so that we can reproduce the issue.

**Reply by Byague** [reply]
Ok, sorry for the screenshot. 

template="Fischer344_nii_v4/Fischer344_template.nii"
template_mask="Fischer344_nii_v4/Fischer344_template_mask.nii"

specimen="AlzheimerTemplate.nii"
specimen_mask="brainmask.nii"

fixed_image="fixed.nii.gz"
moving_image="moving.nii.gz"

ImageMath 3 $fixed_image m $template $template_mask
ImageMath 3 $moving_image m $specimen $specimen_mask

antsRegistrationSyN.sh -f moving.nii.gz -m fixed.nii.gz -o test1 -n 16

antsApplyTransforms -d 3 -i Fischer344_labels_v3.nii -r test1Warped.nii.gz -o finalresult.nii.gz -t test11Warp.nii.gz -t test10GenericAffine.mat -n NearestNeighbor

I reorientate the labels from IPL to LPI in itksnap, but when I shut down and turn on the computer labels change to its original orientation.

Sorry for disturbing with such a simple thing, and thank you so much for your help.

Inputs and outputs: https://we.tl/t-CA9bYPnNrJ

**Reply by cookpa** [reply]
> antsRegistrationSyN.sh -f moving.nii.gz -m fixed.nii.gz -o test1 -n 16

Here you've used "moving.nii.gz" as the fixed image and "fixed.nii.gz" as the moving image.

**Reply by cookpa** [reply, marked as answer]
Also, your labels are not in the same physical space as your fixed image. This is incompatible with registration, the image labels should have the same header (see `CopyImageHeaderInformation`)

If you load fixed into ITK-SNAP and then the labels (load as a grayscale image, NOT a segmentation), they should be aligned. But they are not:

<img width="667" alt="image" src="https://github.com/ANTsX/ANTs/assets/611507/99775c4c-8867-4e91-b46d-031ca192f0b4">

**Reply by cookpa** [reply]
After copying the header info

<img width="536" alt="image" src="https://github.com/ANTsX/ANTs/assets/611507/ee72656d-57bf-43c9-a03f-69f938280027">

**Reply by Byague** [reply]
Thousand of thank you both!! Definitely the headache was not using CopyImageHeaderInfo, after that registration was made great!!

**Comment by cookpa** [comment]
FYI, the Fischer 344 atlas is also available in [templateflow](https://www.templateflow.org/browse/) along with a good brain mask

**Reply by ntustison** [reply]
Thanks @cookpa
